src/Backups/BackupImpl.h

```diff
@@ -114,6 +111,10 @@ class BackupImpl : public IBackup
 
     std::unique_ptr<SeekableReadBuffer> readFileImpl(const SizeAndChecksum & size_and_checksum, bool read_encrypted) const;
 
+    bool fileExists(const SizeAndChecksum & size_and_checksum) const override;
+    std::unique_ptr<SeekableReadBuffer> readFile(const SizeAndChecksum & size_and_checksum) const override;
+    size_t copyFileToDisk(const SizeAndChecksum & size_and_checksum, DiskPtr destination_disk, const String & destination_path, WriteMode write_mode) const override;
```

Comment:
Did you just move these functions to private? Why?

Reply:
That's when I didn't rely on `size` + `checksum` for reference files. References were not accessible using `SizeAndChecksum` + all those overloads were used only internally so I moved them to private to prevent someone accidentally using it.
It still makes sense to me to have it as private as it's only used internally but I can revert the change.

Reply:
All 3 these functions are used only when we read a backup and there are no references at that point.

Reply:
yes, now it's not a problem, it's a leftover from previous implementation